scrapy/pipelines/files.py

```diff
@@ -261,13 +270,19 @@ def from_settings(cls, settings):
         store_uri = settings['FILES_STORE']
         return cls(store_uri, settings=settings)
 
-    def _get_store(self, uri):
+    def _get_store(self, uri, settings):
         if os.path.isabs(uri):  # to support win32 paths like: C:\\some\dir
             scheme = 'file'
         else:
             scheme = urlparse(uri).scheme
         store_cls = self.STORE_SCHEMES[scheme]
-        return store_cls(uri)
+        store = store_cls(uri)
+        # this is a bit ugly but since S3FilesStore does not accept
+        # settings when instantiated, and to keep it backward-ompatible,
+        # the store object is patched with the region name
+        if scheme == 's3':
```

Comment:
Other way would be to implement `from_settings` class method in `S3FilesStore`, in the same way like Scrapy [works](https://github.com/scrapy/scrapy/blob/c22a4e3bb84448f778613717c3749f226df7880f/scrapy/middleware.py#L35-L40) with middlewares, for me it looks a bit weird that `*Store` instances don't have direct access to Scrapy settings, and I'm afraid that amount of such hacks will grow over time. I wonder if it adding such method breaks backward compatibility too though..